src/containers/Transaction/Transaction.js

```diff
@@ -49,27 +60,44 @@ const Transaction = ({ stockExchangeData }) => {
     resolver: yupResolver(schema)
   });
 
-  const [isBuying, currencyName, amount] = [watch('isBuying'), watch('currencyName'), watch('amount')];
+  const history = useHistory();
+  const location = useLocation();
   const dispatch = useDispatch();
+  const { currency: currencyUrlParam } = useParams();
+  const [
+    isBuying, price, currencyName, amount
+  ] = [watch('isBuying'), watch('price'), watch('currencyName'), watch('amount')];
 
-  const onSubmit = (data) => {
-    const { amount, currencyName, isBuying, price } = data;
+  const isCurrencyAvailable = (name) => stockExchangeData.map((currency) => currency.name).includes(name);
 
-    // TODO: this is temporary ID and should be replaced by real user ID
-    dispatch(makeExchange({ userId: 1, amount, currencyName, isBuying, price }));
-  };
+  useEffect(() => {
+    if (currencyName) {
+      const locationWithoutCurrencyName = location.pathname.slice(0, -currencyUrlParam?.length);
+      history.push(`${locationWithoutCurrencyName}${currencyName}`);
+    } else if (isCurrencyAvailable(currencyUrlParam)) {
```

Comment:
It never gets call. And because of that and `DEFAULT_CRYPTO` in `defaultValues` above we cannot enter eg. `/exchange/ETC` directly, we always land on BTC <- that thing should be fixed.

Reply:
We should have tests for that behaviour.